main.go

```diff
@@ -1,235 +1,237 @@
-var pause bool = false
-
-func main() {
-
-	InitializeConfig()
-
-	go LoadAndRunLoop()
-
-	r := mux.NewRouter()
-	r.HandleFunc("/play", PlayHandler)
-	r.HandleFunc("/pause", PauseHandler)
-	r.HandleFunc("/resume", ResumeHandler)
-	r.HandleFunc("/skip", SkipHandler)
-
-	go log.Fatal(http.ListenAndServe(":8080", r))
+// package main
+
+// import (
+// 	"bufio"
+// 	"encoding/csv"
+// 	"fmt"
+// 	"io"
+// 	"io/ioutil"
+// 	"log"
+// 	"net"
+// 	"net/http"
+// 	"os"
+// 	"strings"
+// 	"time"
+
+// 	"github.com/fsnotify/fsnotify"
+// 	"github.com/gorilla/mux"
+// 	"github.com/spf13/viper"
+// )
+
+// type Program struct {
+// 	URL      string
+// 	Duration time.Duration
+// }
+
+// // InitializeConfig loads our configuration using Viper package.
+// func InitializeConfig() {
+
+// 	viper.SetConfigType("yaml")
+// 	// Set config file
+// 	viper.SetConfigName("config")
+
+// 	// Add config path
+// 	//	viper.AddConfigPath("$HOME/.gorotator")
+// 	viper.AddConfigPath(".")
+
+// 	// Read in the config
+// 	err := viper.ReadInConfig() // Find and read the config file
+// 	if err != nil {             // Handle errors reading the config file
+// 		panic(fmt.Errorf("Fatal error config file: %s \n", err))
+// 	}
+
+// 	// Load default settings
+// 	viper.SetDefault("debug", false)
+
+// 	viper.SetEnvPrefix("gorotator") // will be uppercased automatically
+// 	viper.BindEnv("debug")
+// 	viper.BindEnv("ip")
+// 	viper.BindEnv("port")
+
+// 	// Do some flag handling and any complicated config logic
+// 	if !viper.IsSet("ip") || !viper.IsSet("port") {
+// 		fmt.Println("Configuration error.  Both IP and PORT must be set via either config or environment.")
+// 		os.Exit(1)
+// 	}
+
+// 	viper.WatchConfig()
+// 	viper.OnConfigChange(func(e fsnotify.Event) {
+// 		fmt.Println("\nConfig file changed:", e.Name)
+// 		abort <- struct{}{}
+// 		fmt.Printf("Content will change immediately.\n\n")
+// 	})
+
+// }
+
+// // Loads a list of programs.
+// // A program consists of a list things to display on the rotator along
+// // with a number of seconds to display each one before moving on.
+// func loadProgramList(filename string) []Program {
+
+// 	fmt.Printf("Loading programs from %s\n", filename)
+
+// 	var list []Program
+
+// 	bytes, err := ioutil.ReadFile(filename)
+// 	if err != nil {
+// 		log.Fatal(err)
+// 	}
+// 	webpages := string(bytes)
+
+// 	r := csv.NewReader(strings.NewReader(webpages))
+
+// 	for {
+// 		var p Program
+// 		record, err := r.Read()
+// 		if err == io.EOF {
+// 			fmt.Printf("Finished loading programs.\n\n")
+// 			break
+// 		}
+// 		if err != nil {
+// 			log.Fatal(err)
+// 		}
+// 		p.URL = record[0]
+// 		p.Duration, err = time.ParseDuration(record[1])
+// 		fmt.Printf("  Loaded program %.50s to show for %s.\n", p.URL, p.Duration)
+// 		list = append(list, p)
+// 	}
+
+// 	return list
+// }
+
+// func runProgram(program Program) {
+
+// 	// Does this leak goroutines over time because they are created more than they are consumed?
```

Comment:
Thanks for the suggestion for how to fix it!   Later I plan to switch to termbox-go when/if I add more console controls.  For now, I'll move to main and launch in a loop -- it's does read keypresses, but it does trigger when return is pressed and that's good enough for me for now.